lib/client.js

```diff
@@ -29,6 +29,16 @@ const MetricsClient = class MetricsClient extends stream.Transform {
             value: id || crypto.randomBytes(3 * 4).toString('base64'),
             enumerable: true,
         });
+
+        // Avoid hitting the max listeners limit when multiple
+        // streams is piped into the same stream.
+        this.on('pipe', (src) => {
+            this.setMaxListeners(this.getMaxListeners() + 1);
+        });
+
+        this.on('unpipe', () => {
+            this.setMaxListeners(this.getMaxListeners() - 1);
+        });
```

Comment:
How come this works? I thought I understood that the thing being piped into would keep increasing the num listeners. I assumed that meant that it was the consumer that was increasing in listeners all the time.

Reply:
Explained in person.